pkgs/games/unciv/default.nix

```diff
@@ -0,0 +1,60 @@
+{ stdenv
+, lib
+, fetchurl
+, copyDesktopItems
+, makeDesktopItem
+, makeWrapper
+, jre
+, libpulseaudio
+, libXxf86vm
+}:
+let
+  desktopItem = makeDesktopItem {
+    name = "unciv";
+    exec = "unciv";
+    comment = "An open-source Android/Desktop remake of Civ V";
+    desktopName = "Unciv";
+    categories = "Game;";
+  };
+
+  envLibPath = lib.makeLibraryPath [
+    libpulseaudio
+    libXxf86vm
+  ];
+
+in
+stdenv.mkDerivation rec {
+  pname = "unciv";
+
+  version = "3.12.8";
+
+  src = fetchurl {
+    url = "https://github.com/yairm210/Unciv/releases/download/${version}/Unciv.jar";
+    sha256 = "178lasa6ahwg2s2hamm13yysg42qm13v6a9pgs6nm66np93nskc7";
+  };
+
+  phases = "installPhase";
```

Comment:
```suggestion
```

Please do not overwrite phases. Disable the ones you do not want to run.

Reply:
Why? How to disable unwanted phase? Do you have link to doc describing this? I just used what I saw in existing jre packages back then. Isvthere uptodate coding standard for nix packages?

Reply:
> Why?

Because certain hooks require certain phases and this is a big anti pattern.

> How to disable unwanted phase? 

```
doUnpack = false;
doInstall = false;
doCheck = false;
```

> Do you have link to doc describing this?

I don't actually read the docs but use other packages and other peoples reviews as reference. Can't link you anything.

> I just used what I saw in existing jre packages back then.

Yeah, they are using outdated patterns.

> Is there uptodate coding standard for nix packages?

It is constantly changing. We recently deprecated stdenv.lib and pkgconfig -> pkg-config.

Reply:
Even with doUnpack = false it complains:

```
building '/nix/store/kzydfc3k8va6swhgr5pbajbggijlyzz8-unciv-3.12.8.drv'...
unpacking sources
unpacking source archive /nix/store/xkimqn9978bpfy9bf0cnx0bk7blcygk1-Unciv.jar
do not know how to unpack source archive /nix/store/xkimqn9978bpfy9bf0cnx0bk7blcygk1-Unciv.jar
```

Check whole nixpkgs, no one single package has doUnpack = false; dunno why...

Reply:
I am not that great with remembering the exact names. The right name is dontUnpack, dontInstall and doCheck for Python. I would suggest to ripgrep a bit through the source and you will find good examples.

Reply:
Thanks, the dontUnpack works.